mm2src/coins/eth.rs

```diff
@@ -752,12 +755,13 @@ impl SwapOps for EthCoin {
         _htlc_privkey: &[u8],
         swap_contract_address: &Option<BytesJson>,
     ) -> TransactionFut {
-        let tx: UnverifiedTransaction = try_fus!(rlp::decode(taker_payment_tx));
-        let signed = try_fus!(SignedEthTx::new(tx));
-        let swap_contract_address = try_fus!(swap_contract_address.try_to_address());
+        let tx: UnverifiedTransaction = try_tx_fus!(rlp::decode(taker_payment_tx));
+        let signed = try_tx_fus!(SignedEthTx::new(tx));
+        let swap_contract_address = try_tx_fus!(swap_contract_address.try_to_address());
```

Comment:
What do you think, should we return `TransactionErr::TxRecoverableError` if `swap_contract_address.try_to_address()` fails?
Please also check if we should do that in other places.

Reply:
I think we should, why not if we can just return it right?